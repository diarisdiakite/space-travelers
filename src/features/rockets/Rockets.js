import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import Badge from 'react-bootstrap/Badge';
import { reserveAction } from './rocketsSlice';
import '../../assets/css/rockets.css';
import imageError from '../../assets/images/image_not_available.png';

const Rockets = ({
  id, rocketName, description, flickrImages, reserved = false,
}) => {
  const dispatch = useDispatch();

  return (
    <div className="main-container">
      <div className="rocket-card">
        <div className="rocket-image-container">
          <img onError={(e) => { e.target.src = imageError; }} src={flickrImages} alt="rocket" />
        </div>
        <div className="rocket-content-container">
          <h2 key={id}>{rocketName}</h2>
          <p>
            {reserved && (
              <Badge bg="success">Reserved</Badge>
            )}
            <span>{description}</span>
          </p>
          {reserved && (
            <button type="button" className="cancel-button" onClick={() => dispatch(reserveAction(id))}>Cancel Reservation</button>
          )}
          {!reserved && (
            <button type="button" className="reserve-button" onClick={() => dispatch(reserveAction(id))}>Reserve Rocket</button>
          )}
        </div>
      </div>
    </div>
  );
};

Rockets.propTypes = {
  id: PropTypes.string.isRequired,
  rocketName: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  flickrImages: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};
export default Rockets;
