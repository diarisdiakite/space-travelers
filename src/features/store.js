import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missionsSlice';
import usersReducer from './users/usersSlice';
import rocketsSlice from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    users: usersReducer,
    rockets: rocketsSlice,
  },
});

export default store;
