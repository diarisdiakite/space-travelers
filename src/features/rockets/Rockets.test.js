import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import store from '../store';
import Rockets from './Rockets';
import imageError from '../../assets/images/image_not_available.png';

describe('Rockets component', () => {
  it('renders correctly', () => {
    const data = [
      {
        id: '5e9d0d95eda69955f709d1eb',
        name: 'Falcon 1',
        type: 'rocket',
        flickr_images: imageError,
        description: 'The Falcon 1 was an expendable launch system privately',
        reserved: false,
      },
    ];
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Rockets
              id={data[0].id}
              rocketName={data[0].name}
              description={data[0].description}
              flickrImages={data[0].flickr_images}
              reserved={data[0].reserved}
            />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
