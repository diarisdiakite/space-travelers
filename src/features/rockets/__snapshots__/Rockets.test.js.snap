// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component renders correctly 1`] = `
<div
  className="main-container"
>
  <div
    className="rocket-card"
  >
    <div
      className="rocket-image-container"
    >
      <img
        alt="rocket"
        onError={[Function]}
        src="image_not_available.png"
      />
    </div>
    <div
      className="rocket-content-container"
    >
      <h2>
        Falcon 1
      </h2>
      <p>
        <span>
          The Falcon 1 was an expendable launch system privately
        </span>
      </p>
      <button
        className="reserve-button"
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</div>
`;
